### PR TITLE
Maui.Options

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**

### CODE REVIEW

**Comment by VladislavAntonyuk** [comment]
@brminnick could you please take a look?

File: docs/maui/options.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Allows to return default value instead of throwing an exception when using `BaseConverter{TFrom,TTo}`. Default value is false.
+
+## SetShouldSuppressExceptionsInAnimations
+
+Allows to return default value instead of throwing an exception when using `AnimationBehavior`. Default value is false.
+
+## SetShouldSuppressExceptionsInBehaviors
+
+Allows to return default value instead of throwing an exception when using `BaseBehavior{TView}`. Default value is false.
```
Comment (brminnick) [inline docs/maui/options.md:22]:
```suggestion
---
title: CommunityToolkit.Maui.Options
author: VladislavAntonyuk
description: CommunityToolkit.Maui.Options allows to customize the CommunityToolkit.Maui.
ms.date: 09/14/2022
---

# CommunityToolkit.Maui.Options

`CommunityToolkit.Maui.Options` allows to customize the `CommunityToolkit.Maui`. 

`CommunityToolkit.Maui.Options` allows developers to customize `CommunityToolkit.Maui`. 

`Options` should be assigned at startup when calling `.UseMauiCommunityToolkit()`:

```cs
var builder = MauiApp.CreateBuilder();
builder.UseMauiCommunityToolkit(options =>
{
    options.SetShouldSuppressExceptionsInConverters(false);
    options.SetShouldSuppressExceptionsInBehaviors(false);
    options.SetShouldSuppressExceptionsInAnimations(false);
})
```

## SetShouldSuppressExceptionsInConverters

When set to `true`, if a converter implementing     `CommunityToolkit.Maui.Converters.BaseConverter` throws an `Exception`, the `Exception` will be caught, logged via `Debug.WriteLine()`, and a predetermined default value will be returned.

### Example
This option is enabled when calling `.UseMauiCommunityToolkit()`:

```cs
var builder = MauiApp.CreateBuilder();
builder.UseMauiCommunityToolkit(options =>
{
    options.SetShouldSuppressExceptionsInConverters(true);
})
```

### Default Return Values
When set to `true`, a default value will be returned when a Converter throws an `Exception`.

Two default values are included:
- `public object? ICommunityToolkitValueConverter.DefaultConvertReturnValue { get; set; }`
  - Default value returned when `Convert(object? value, Type targetType, object? parameter, CultureInfo? culture)` throws an `Exception`
- `public object ICommunityToolkitValueConverter.DefaultConvertBackReturnValue { get; set; }`
  - Default value returned when `ConvertBack(object? value, Type targetType, object? parameter, CultureInfo? culture)` throws an `Exception`

Here is an example setting the default values for  `BoolToObjectConverter`:

#### XAML

```xml
<ContentPage.Resources>
    <SolidColorBrush x:Key="TrueColorBrush">Green</SolidColorBrush>
    <SolidColorBrush x:Key="FalseColorBrush">Red</SolidColorBrush>
    <mct:BoolToObjectConverter x:Key="BoolToColorBrushConverter" 
                                TrueObject="{StaticResource TrueColorBrush}" 
                                FalseObject="{StaticResource FalseColorBrush}"
                                DefaultConvertReturnValue="{StaticResource FalseColorBrush}"
                                DefaultConvertBackReturnValue="False"/>
</ContentPage.Resources>
```

#### C#

```cs
var boolToColorBrushConverter = new BoolToObjectConverter
{
    TrueObject = new SolidColorBrush(Colors.Green),
    FalseObject = new SolidColorBrush(Colors.Red),
    DefaultConvertReturnValue = new SolidColorBrush(Colors.Red),
    DefaultConvertBackReturnValue = false
};
```

## SetShouldSuppressExceptionsInAnimations

When set to `true`, if an Animation implementing     `CommunityToolkit.Maui.Behaviors.AnimationBehavior` throws an `Exception`, the `Exception` will be caught and will be logged via `Debug.WriteLine()`.

### Example
This option is enabled when calling `.UseMauiCommunityToolkit()`:

```cs
var builder = MauiApp.CreateBuilder();
builder.UseMauiCommunityToolkit(options =>
{
    options.SetShouldSuppressExceptionsInAnimations(true);
})
```

## SetShouldSuppressExceptionsInBehaviors

When set to `true`, if a Behavior implementing     `CommunityToolkit.Maui.Behaviors.BaseBehavior` throws an `Exception`, the `Exception` will be caught and will be logged via `Debug.WriteLine()`.

### Example
This option is enabled when calling `.UseMauiCommunityToolkit()`:

```cs
var builder = MauiApp.CreateBuilder();
builder.UseMauiCommunityToolkit(options =>
{
    options.SetShouldSuppressExceptionsInBehaviors(true);
})
```
```

**Review by brminnick — CHANGES_REQUESTED** [review]
Hey Vlad! What do you think about adding more details to `Options.md`?

File: docs/maui/options.md (hunk 1 above)
Comment (VladislavAntonyuk) [reply docs/maui/options.md:22]: Agree, I mentioned it in get-started.md like advanced settings. It make sense to provide more details here. Just don't want to add a lot of info in get-started

File: docs/maui/options.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+title: CommunityToolkit.Maui.Options
```
Comment (jfversluis) [inline docs/maui/options.md:2]:
```suggestion
title: Options - .NET MAUI Community Toolkit
```

**Review by jfversluis — APPROVED** [review]

**Pull request merged**